src/uct/ib/rc/accel/rc_mlx5.inl

```diff
@@ -413,15 +413,22 @@ uct_rc_mlx5_iface_common_am_handler(uct_rc_mlx5_iface_common_t *iface,
         rc_ops = ucs_derived_of(iface->super.super.ops, uct_rc_iface_ops_t);
 
         /* coverity[overrun-buffer-val] */
-        status = rc_ops->fc_handler(&iface->super, qp_num, &hdr->rc_hdr,
-                                    byte_len - sizeof(*hdr),
-                                    cqe->imm_inval_pkey, cqe->slid, flags);
-    } else {
-        status = uct_iface_invoke_am(&iface->super.super.super, hdr->rc_hdr.am_id,
-                                     hdr + 1, byte_len - sizeof(*hdr),
-                                     flags);
+        rc_ops->fc_handler(&iface->super, qp_num, &hdr->rc_hdr,
+                           byte_len - sizeof(*hdr), cqe->imm_inval_pkey,
+                           cqe->slid, flags);
+        if (uct_rc_fc_get_fc_hdr(hdr->rc_hdr.am_id) ==
+                    UCT_RC_EP_FC_PURE_GRANT) {
```

Comment:
the reason to put it inside fc_handler was to reduce code footprint of the fast-path polling function